### PR TITLE
db: Prune all debug file caches (symcaches and CFI caches)

### PR DESCRIPTION
**Pull request by jan-auer**
Previous versions of symbolic emitted invalid SymCaches and CFI caches under certain circumstances. This PR contains a migration to prune all of them, which forces them to be recreated once requested during processing. 

**The migration will be run with custom raw queries in production.**

### CODE REVIEW

**Comment by mitsuhiko** [comment]
I guess that's fine? As per conversation earlier i think we should try not to do it like this going forward (just a version bump) and make sure this also works for CFI caches if it does not do that yet.

**Review by mitsuhiko — APPROVED** [review]

File: src/sentry/south_migrations/0476_truncate_symcaches_and_cficaches.py [hunk 1]
```diff
@@ -0,0 +1,1328 @@
...
+
+        try:
+            last_cficache_id = ProjectCfiCacheFile.objects \
+                .values_list('cache_file_id') \
+                .latest('id')[0]
+        except ProjectCfiCacheFile.DoesNotExit:
+            last_symcache_id = None
+
+        if last_symcache_id is not None:
+            ProjectSymCacheFile.objects.all().delete()
```
Comment (mattrobenolt) [inline src/sentry/south_migrations/0476_truncate_symcaches_and_cficaches.py]: How many of these are there?

File: src/sentry/south_migrations/0476_truncate_symcaches_and_cficaches.py [hunk 2]
```diff
@@ -0,0 +1,1328 @@
...
+        if last_cficache_id is not None:
+            ProjectCfiCacheFile.objects.all().delete()
+
+        if last_symcache_id is not None:
+            File.objects\
```
Comment (mattrobenolt) [inline src/sentry/south_migrations/0476_truncate_symcaches_and_cficaches.py]: Same here. These are potentially painful queries to do just at once. How many rows are we talking about here? We also don't have an index on this, so this seems like a really bad query.

File: src/sentry/south_migrations/0476_truncate_symcaches_and_cficaches.py (hunk 1 above)
Comment (mitsuhiko) [reply src/sentry/south_migrations/0476_truncate_symcaches_and_cficaches.py]: The intention that this only runs for on-prem customers.

**Comment by mitsuhiko** [comment]
Can we dump the raw query we're going to run into this PR somewhere so it can be reviewed as well?

File: src/sentry/south_migrations/0476_truncate_symcaches_and_cficaches.py (hunk 1 above)
Comment (mattrobenolt) [reply src/sentry/south_migrations/0476_truncate_symcaches_and_cficaches.py]: Oh, I saw it was marked as is_dangerous, but we don't need to run any of this at all for us?

Comment (jan-auer) [reply src/sentry/south_migrations/0476_truncate_symcaches_and_cficaches.py]: 1.6M and 270k

Comment (jan-auer) [reply src/sentry/south_migrations/0476_truncate_symcaches_and_cficaches.py]: We'd run a `TRUNCATE` for this one in prod, and query on `sentry_file` would just touch cold rows after that.

**Review by tonyo — APPROVED** [review]
lgtm for on-prem, we can have a separate discussion for saas.

**Pull request closed without merging**